### PR TITLE
fix: use ignore_index for user-turn marker token label in fine-tuning

### PR DESCRIPTION
**Pull request by Mr-Neutr0n**
## Summary

In `fine-tune/fine-tune.py`, the `preprocessing` method incorrectly sets the label for the user-turn marker token (token 195) to `eos_token_id` instead of `IGNORE_INDEX` (-100).

```python
# Before (bug):
if from_ == "human":
    input_ids += self.user_tokens + value_ids
    labels += [self.tokenizer.eos_token_id] + [self.ignore_index] * len(value_ids)
    #          ^^^ Should be self.ignore_index

# After (fix):
if from_ == "human":
    input_ids += self.user_tokens + value_ids
    labels += [self.ignore_index] + [self.ignore_index] * len(value_ids)
```

The special user-turn marker token (195) is a structural delimiter, not a token the model should learn to predict. Using `eos_token_id` as its label trains the model to predict EOS at every human-turn boundary, which degrades multi-turn conversation quality. The label should be `IGNORE_INDEX` (-100) so that it is excluded from the cross-entropy loss computation, consistent with how the assistant-turn marker token is already handled on the `else` branch.

### CODE REVIEW

**Comment by Mr-Neutr0n** [comment]
Closing this one to tidy up my open pull requests.

It's been open around five months with no review activity, and I'd rather withdraw it than leave it sitting in your queue. Nothing needed from you, and no hard feelings at all.

If the fix is still wanted, this can be reopened, or I'm happy to redo it properly against current main. Apologies for the noise.
